### PR TITLE
Added barrier::increment function to increase total number of thread

### PR DESCRIPTION
**Pull request by tianyi93**
Added barrier:: increment function to increase the total number of thread
Test added

### CODE REVIEW

File: hpx/lcos/local/barrier.hpp [hunk 1]
```diff
@@ -46,9 +46,10 @@ namespace hpx { namespace lcos { namespace local
         /// releasing all waiting threads as soon as the last \a thread
         /// entered this function.
         void wait();
+        void increase();
```
Comment (hkaiser) [inline hpx/lcos/local/barrier.hpp]: Let's align the name of this function with previous art. The closest would probably `std::latch::count_down()`. I'd suggest to rename this to `barrier::count_up()`. Would that work for everybody?

File: src/lcos/local/local_barrier.cpp [hunk 2]
```diff
@@ -67,5 +67,9 @@ namespace hpx { namespace lcos { namespace local
             }
         }
     }
+    void barrier::increase(){
+        std::unique_lock<mutex_type> l(mtx_);
+        number_of_threads_+=1;
+    }
```
Comment (hkaiser) [inline src/lcos/local/local_barrier.cpp]: Please format your code to match the formatting of the surrounding code.

File: src/lcos/local/local_barrier.cpp [hunk 3]
```diff
@@ -67,5 +67,9 @@ namespace hpx { namespace lcos { namespace local
             }
         }
     }
+    void barrier::increase(){
+        std::unique_lock<mutex_type> l(mtx_);
+        number_of_threads_+=1;
```
Comment (hkaiser) [inline src/lcos/local/local_barrier.cpp]: `++number_of_threads_;` would probably be more readable.

**Comment by tianyi93** [comment]
@hkaiser I did those changes based on the comments, it seems that still some checks are failing, please let me know if there is anything I should improve, thanks

**Pull request merged**